uwblueprint/starter-code-v2
# Python Postgres CRUD

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Entity-CRUD-service-using-PostgreSQL-SQLAlchemy-f9848a4b43f540ef90228fbd02a67484, https://www.notion.so/uwblueprintexecs/Entity-CRUD-REST-endpoints-Flask-089178d3b98140c2bab99f5932a48340)



## Implementation description
* Essentially building off of what was previously made in the pfrp starter code



## Steps to test
1. docker-compose up --build the project, comment out ts-backend portion of the docker-compose.yml
2. Make a GET request to localhost:5000/entities, verify that the output is   ```[ ] ```
3. Make a POST request to localhost:5000/entities, with the JSON as 
  ```
{
    "string_field": "Joe",
    "int_field": 21,
    "string_array_field": ["C"],
    "enum_field": "B",
    "bool_field": true
}
```
Verify that the output matches after making another GET request to localhost:5000/entities and contains an id
4. Make a GET request to localhost:5000/entities/id replacing id with the id of the newly created object. Verify that the object that you recently created is the same.
5. Make a PUT request to localhost:5000/entities/id replacing id with the id of the newly created object, with the newly updated fields as 
  ```
{
    "string_field": "Jane",
    "int_field": 42,
    "string_array_field": ["D"],
    "enum_field": "C",
    "bool_field": false
}
```
Verify that the output matches after making another GET request to localhost:5000/entities/id with the id of the updated object

6. Make a DELETE request to localhost:5000/entities/id with the id of the object. Verify that a GET request to localhost:5000  returns []
7. If all steps work, CRUD functionality for python should be good!


## What should reviewers focus on?
*  Endpoints in routes
*  Validation from entity_dto.py in resources
*  entity.py


## Checklist
- [* ] My PR name is descriptive and in imperative tense
- [* ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [* ] I have run the appropriate linter(s)
- [ *] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
